policy/modules/admin/bootloader.te

```diff
@@ -82,6 +84,8 @@ dev_dontaudit_rw_generic_dev_nodes(bootloader_t)
 dev_read_rand(bootloader_t)
 dev_read_urand(bootloader_t)
 dev_read_sysfs(bootloader_t)
+# for updating efi variables
+dev_write_sysfs(bootloader_t)
```

Comment:
that doesnt make sense, efi variables are on efivarfs and not sysfs

Reply:
The comment is probably wrong, and I'm not really sure what to make of the actual denial:
```
time->Sun Mar 14 16:42:55 2021
type=PROCTITLE msg=audit(1615754575.219:5887): proctitle=656669626F6F746D6772002D71002D620030303030002D42
type=PATH msg=audit(1615754575.219:5887): item=0 name="/sys/firmware/efi/vars/del_var" inode=20936 dev=00:0e mode=0100200 ouid=0 ogid=701 rdev=00:00 obj=system_u:object_r:sysfs_t:s0 nametype=NORMAL cap_fp=0 cap_fi=0 cap_fe=0 cap_fver=0 cap_frootid=0
type=CWD msg=audit(1615754575.219:5887): cwd="/root"
type=SYSCALL msg=audit(1615754575.219:5887): arch=c000003e syscall=257 success=no exit=-13 a0=ffffff9c a1=7b17a7cc9f60 a2=1 a3=0 items=1 ppid=228501 pid=228523 auid=1000 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=pts0 ses=11 comm="efibootmgr" exe="/usr/sbin/efibootmgr" subj=staff_u:sysadm_r:bootloader_t:s0-s0:c0.c1023 key="access"
type=AVC msg=audit(1615754575.219:5887): avc:  denied  { write } for  pid=228523 comm="efibootmgr" name="del_var" dev="sysfs" ino=20936 scontext=staff_u:sysadm_r:bootloader_t:s0-s0:c0.c1023 tcontext=system_u:object_r:sysfs_t:s0 tclass=file permissive=0
```

Reply:
/sys/firmware/efi/vars might just be mislabeled, I guess that depends on the context in fstab

Reply:
Hmm. Even in Fedora's policy these are `sysfs_t`. I'm thinking I will drop this access for now until I can investigate further.

Reply:
efivars is mounted so fedora might have the sysfs_t label on it, but mount it with context=efivarfs (I'd have to check and dont have a EFI fedora VM rn)

Reply:
I see `genfscon efivarfs /  system_u:object_r:efivarfs_t:s0` in the Fedora policy, and it looks like something we need in refpolicy.

Reply:
OK. I can add that here.

Reply:
But it seems that's already in refpolicy? https://github.com/SELinuxProject/refpolicy/blob/master/policy/modules/kernel/filesystem.te#L106

Reply:
My mistake, I must have looked at the wrong branch.

Gentoo isn't using efivarfs?

Reply:
@perfinion do you have any input for this thread?

Reply:
I should clarify: this was added because in testing I found that `grub-mkconfig` would try to write to files in `/sys/firmware/efi/varfs` which is `sysfs_t`. Apparently this is different from `/sys/firmware/efi/efivars` which is actually where the `efivarfs` filesystem lives. I found that this is also the case on RHEL. Not sure what the implications of that are.

Reply:
Ah. The sysfs entries is the old way. efivarfs is the new way. I see this in grub:
```
  /*
   * Linux uses efivarfs (mounted on /sys/firmware/efi/efivars) to access the
   * EFI variable store. Some legacy systems may still use the deprecated
   * efivars interface (accessed through /sys/firmware/efi/vars). Where both
   * are present, libefivar will use the former in preference, so attempting
   * to load efivars will not interfere with later operations.
   */
```
The change is over 2 years old, so It seems this sysfs access should be dontaudited. Or if Gentoo doesn't have a new enough grub, then the sysfs write access should live in the Gentoo policy.

Reply:
I'm sure that Gentoo has a new enough Grub, so maybe this is some sort of fallback behavior that I am triggering somehow in my environment. Either way, the above change isn't in this PR anymore, but I can `dontaudit` it as you say.